constraints.pro

```diff
@@ -0,0 +1,58 @@
+  workspace_has_dependency(WorkspaceCwd, DependencyIdent, DependencyRange, 'devDependencies'),
+  % Only consider dependency ranges that start with 'workspace:'
+  atom_concat('workspace:', _, DependencyRange).
+
+% Enforces the license in all public workspaces while removing it from private workspaces
+gen_enforced_field(WorkspaceCwd, 'license', 'MIT') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'license', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% Enforces the repository field for all public workspaces while removing it from private workspaces
+gen_enforced_field(WorkspaceCwd, 'repository.type', 'git') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'repository.url', 'https://github.com/babel/babel.git') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'repository.directory', WorkspaceCwd) :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'repository', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% Enforces 'publishConfig.access' is set to public for public workspaces while removing it from private workspaces
+gen_enforced_field(WorkspaceCwd, 'publishConfig.access', 'public') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
+gen_enforced_field(WorkspaceCwd, 'publishConfig.access', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% Enforces the engines.node field for all workspaces except '@babel/eslint*'
+gen_enforced_field(WorkspaceCwd, 'engines.node', '>=6.9.0') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true),
+  % Get the workspace name
+  workspace_ident(WorkspaceCwd, WorkspaceIdent),
+  % Exempt from the rule as it supports '>=4'. TODO: remove with the next major
+  WorkspaceIdent \= '@babel/plugin-proposal-unicode-property-regex',
+  % Exempt from the rule as it supports '>=6.0.0'. TODO: remove with the next major
+  WorkspaceIdent \= '@babel/parser',
+  % Skip '@babel/eslint*' workspaces. TODO: remove with the next major
+  \+ atom_concat('@babel/eslint', _, WorkspaceIdent).
+
+% Enforces the engines.node field for '@babel/eslint*' workspaces
+% TODO: remove with the next major
+gen_enforced_field(WorkspaceCwd, 'engines.node', '^10.13.0 || ^12.13.0 || >=14.0.0') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true),
+  % Get the workspace name
+  workspace_ident(WorkspaceCwd, WorkspaceIdent),
+  % Only target '@babel/eslint*' workspaces
+  atom_concat('@babel/eslint', _, WorkspaceIdent).
+
+% Removes the 'engines.node' field from private workspaces
+gen_enforced_field(WorkspaceCwd, 'engines.node', null) :-
+  workspace_field(WorkspaceCwd, 'private', true).
+
+% Enforces the author field to be consistent
+gen_enforced_field(WorkspaceCwd, 'author', 'The Babel Team (https://babel.dev/team)') :-
+  \+ workspace_field(WorkspaceCwd, 'private', true).
```

Comment:
While I would love to do it, I'm not 100% sure that we can legally remove Sebastian and Logan from the existing `author` fields.

Maybe we should add an explicit list of packages that can have a different author (it doesn't need to be maintained anyway, it's "legacy"), and only enforce `The Babel Team` for new packages.

We can probably replace @hzoo in `@babel/preset-env` if he agrees, since he's still a team member.

Reply:
While I'm not a lawyer the MIT license only requires that the license is included so based on that it's fine to change it as it's part of the "Software" and permission was granted to modify it.

Reply:
Not a lawyer here either, but I worked a bit on licenses in the past. I just want to mention that license and author rights are two different things.

I _think_ the change here is fine as long as Sebastian and others' attributions are still present otherwise (an AUTHORS file, or maybe the output `git blame` itself is already good enough in this case as long as their contributions are properly labeled in the git repository).

Other free software sometimes uses a CLA that enforces that every contribution's author rights are given to one single organization... but even this doesn't work in some countries (eg: France).

My (unwanted) 2 cents :-)